numpyro/infer/autoguide.py

```diff
@@ -11,7 +11,7 @@
 
 import jax
 from jax import grad, hessian, lax, random, tree_map
-from jax.experimental import stax
+from jax.example_libraries import stax
```

Comment:
Hi @MarcoGorelli, how about checking jax version before importing? (e.g. colab is using an old jax version so it would be nice to still support it). Kind of
```
if jax.__version__ >= "...":
    from jax.example_libraries import stax
else:
    from jax.experimental import stax
```

Reply:
Yes, that's better, thanks!

Should there be a separate CI job to check the oldest supported jax version, or is a local check I've done OK?

Reply:
I think we might skip it. Btw, can we avoid using extra package `packaging`? `jax` follows standard names so we can just compare the strings.

Reply:
Wouldn't comparing strings bring problems if they released, say, `0.2.100`?
```python
>>> '0.2.100' > '0.2.13'
False
```

And sure, the `versiontuple` answer looks good enough for this use-case, it seems very unlikely that JAX would suddenly change the format of their version numbers

Reply:
I think we can safely remove those version checks after a few jax releases or the next numpyro release but I agree the safe solution is better.